sound/Music.java

```diff
@@ -192,7 +192,15 @@ public void playChord(Chord chord, int defaultDuration) {
      * Play a tune.
      * @param tune the tune to play
      */
-    public void playTune(Tune tune) {
+    public void playTune(Tune tune, ImgProvider imp) {
+        int heightTracker = 0;
+        ImgProvider improvider = imp;
+        int[] pixels = improvider.getPix();
+        int height = improvider.getPixHeight();
+        int width = improvider.getPixWidth();
+        int oldRow = 0;
```

Comment:
What are the intents of oldRow and newRow?  How are they used?  The variable names themselves are insufficient to explain their meaning or usage.